sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/_connection.py

```diff
@@ -78,10 +83,12 @@ def __init__(self, endpoint, **kwargs):
         self.state = None  # type: Optional[ConnectionState]
 
         transport = kwargs.get('transport')
+        self._transport_type = kwargs.pop('transport_type')
         if transport:
             self._transport = transport
         elif 'sasl_credential' in kwargs:
-            self._transport = SASLTransport(
+            func = SASLWithWebSocket if self._transport_type is TransportType.AmqpOverWebsocket else SASLTransport
+            self._transport = func(
```

Comment:
if http proxy is passed, we should also use `SASLWithWebSocket`.

also I think we could change to a better name:
```suggestion
            sasl_transport = SASLWithWebSocket if (self._transport_type is TransportType.AmqpOverWebsocket or kwargs.get("http_proxy") else SASLTransport
            self._transport = sasl_transport(
```